### PR TITLE
permissions-center: remove feature flag and old perms syncing code.

### PR DESCRIPTION
**Pull request by sashaostrikov**
This commit only removes the code without moving it around for simplicity. Moving the code closer to `perms_syncer_worker` and deleting `perms_syncer.go` file will be done in a following commit for the sake of minimizing PR review complexity.

Test plan:
Local sg run and confirming that permissions are being synced.
+ main dry run

Part of https://github.com/sourcegraph/sourcegraph/issues/46560

### CODE REVIEW

File: cmd/repo-updater/repoupdater/server.go [hunk 1]
```diff
@@ -221,7 +212,7 @@ func (s *Server) respond(w http.ResponseWriter, code int, v any) {
 			s.Logger.Error("response value error", log.Error(val))
 			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
 			w.WriteHeader(code)
-			fmt.Fprintf(w, "%v", val)
+			_, _ = fmt.Fprintf(w, "%v", val)
```
Comment (kopancek) [inline cmd/repo-updater/repoupdater/server.go:215]: Why is this needed?

File: enterprise/cmd/repo-updater/shared/shared.go [hunk 2]
```diff
@@ -91,14 +70,11 @@ func EnterpriseInit(
...
-
-	go startBackgroundPermsSync(ctx, permsSyncer, db)
-
-	return map[string]debugserver.Dumper{"repoPerms": permsSyncer}, enqueueRepoPermsJob
+	go watchAuthzProviders(ctx, db)
 }
 
-// startBackgroundPermsSync sets up background permissions syncing.
-func startBackgroundPermsSync(ctx context.Context, syncer *authz.PermsSyncer, db ossDB.DB) {
+// watchAuthzProviders sets up background permissions syncing.
```
Comment (kopancek) [inline enterprise/cmd/repo-updater/shared/shared.go:76]: Should we also change the docstring?
```suggestion
// watchAuthzProviders sets up authz providers if config changes
```

**Review by kopancek — APPROVED** [review]
Love PRs that remove code 👍

File: enterprise/cmd/repo-updater/shared/shared.go (hunk 2 above)
Comment (sashaostrikov) [reply enterprise/cmd/repo-updater/shared/shared.go:76]: nice catch, thanks!

**Review by pjlast — APPROVED** [review]
Let us proceed
🎩 
🧐
